helixbass/tsc-rust
# "Type type" macro

In this PR:
- use a new `#[type_type]` procedural macro for `Type` enum variants similar to how `#[ast_type]` macro is used for `Node` variants

Not in this PR:
- updating places that wrap `Type`'s in `Rc`'s to use the generated `.into()` (which encapsulates setting the `_type_wrapper` field on the `BaseType`) and changing function parameters to prefer being `&Type` instead of `Rc<Type>`

To test:
Behavior should still be the same as of #32 

Based on `type-check-variable-usage`
